src/pg_cron.c

```diff
@@ -475,6 +476,133 @@ cron_error_severity(int elevel)
 	return elevel_char;
 }
 
+/*
+ * TIMEZONE
+ */
+
+/*
+ * check_cron_timezone: GUC check_hook for cron.timezone
+ */
+static bool
+check_cron_timezone(char **newval, void **extra, GucSource source)
```

Comment:
can't we just use check_timezone directly? it's declared in commands/variable.h

Reply:
@marcocitus Hello, thank you very much for your review. The reason why we cannot use the check_timezone function here is: for the guc parameter in the extension, a warning is only printed when starting the service, but if the value of cron.timezone is an illegal value Will affect the subsequent ShouldRunTask function.

That is to say: if there is a wrong value here, we want it to report an error, just like timezone, rather than a warning.

![image](https://user-images.githubusercontent.com/49380232/207205279-7c8156fd-2445-40da-bf3e-96dc2acd7b87.png)


Please have a look at the description in this link:

https://github.com/citusdata/pg_cron/pull/219

Reply:
No, I think warning is right and a general postgres policy. The value gets checked on server start, and error would mean the server does not start.